atmo/jobs/views.py

```diff
@@ -133,3 +136,42 @@ def download_spark_job(request, id):
     )
     response['Content-Length'] = spark_job.notebook_s3_object['ContentLength']
     return response
+
+
+@login_required
+@view_permission_required(SparkJob)
+def run_spark_job(request, id):
+    spark_job = SparkJob.objects.get(pk=id)
```

Comment:
Ah, wait, the 404 is already handled by the view_permission_required decorator, which is why there is no need to do that. I should probably add an option to make it clear that this is what is happening.